### PR TITLE
Feat/bulk/all metrics

### PR DESCRIPTION
**Pull request by gweiying**
Adds the following metrics for bulk QR code generation and bulk link creation:
- [x] bulk create count
- [x] bulk validation error (and error types tags)
- [x] custom query strings to track source for s3 bucket download

After release, to add S3 bucket logs and DD streaming for the 3 environments

### CODE REVIEW

File: src/server/modules/bulk/BulkController.ts [hunk 1]
```diff
@@ -59,12 +63,12 @@ export class BulkController {
...
       await this.urlManagementService.bulkCreate(userId, urlMappings, tags)
     } catch (e) {
-      dogstatsd.increment('bulk.hash.failure', 1, 1)
+      dogstatsd.increment(BULK_CREATE_FAILURE, 1, 1)
       res.badRequest(jsonMessage('Something went wrong, please try again.'))
       return
     }
-
-    dogstatsd.increment('bulk.hash.success', 1, 1)
+    dogstatsd.increment(BULK_CREATE_COUNT, urlMappings.length)
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/server/modules/bulk/BulkController.ts]: [730378f](https://github.com/opengovsg/GoGovSG/pull/2134/commits/730378f74c7685e12d2c94de215d5f42552506d3)

File: src/server/util/dogstatsd.ts [hunk 2]
```diff
@@ -18,6 +18,22 @@ export const SHORTLINK_CREATE_TAG_IS_FILE = 'isfile'
 export const SHORTLINK_CREATE_TAG_SOURCE = 'source'
 export const USER_NEW = 'user.new'
 
+export const BULK_VALIDATION_ERROR = 'bulk.validation.error'
+export const BULK_VALIDATION_ERROR_TAGS = {
+  hasUrls: 'errorType:emptyCsv',
+  validHeader: 'errorType: invalidHeader',
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/server/util/dogstatsd.ts]: [e763961](https://github.com/opengovsg/GoGovSG/pull/2134/commits/e7639615edf6e3fca5d2202e734976a1e606c50e)

File: src/server/util/dogstatsd.ts [hunk 3]
```diff
@@ -18,6 +18,22 @@ export const SHORTLINK_CREATE_TAG_IS_FILE = 'isfile'
...
+  acceptableLinkCount: 'errorType:exceededLinkCount',
+  onlyOneColumn: 'errorType:moreThanOneColumn',
+  isNotEmpty: 'errorType:emptyRow',
+  isValidUrl: 'errorType:invalidUrl',
+  isNotBlacklisted: 'errorType:blacklistedUrl',
+  isNotCircularRedirect: 'errorType:circularRedirect',
+  noParsingError: 'errorType:parsingError',
+}
```
[the remark this answers is not in the document]
Comment (gweiying) [reply src/server/util/dogstatsd.ts]: [e763961](https://github.com/opengovsg/GoGovSG/pull/2134/commits/e7639615edf6e3fca5d2202e734976a1e606c50e)

[the remark this answers is not in the document]
Comment (gweiying) [reply src/server/util/dogstatsd.ts]: Thanks for the heads up, didn't know that

File: src/server/modules/bulk/BulkController.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (halfwhole) [reply src/server/modules/bulk/BulkController.ts]: oh whoops i didn't realise it was already in there! looks good

**Review by halfwhole — APPROVED** [review]

**Pull request merged**